### DISCUSSION TITLE
Cannot add to firefox search engines, invalid format error.

### DISCUSSION THREAD

**Discussion by silmarine** [searxng/searxng#3109] [Q&A] [answered]
When trying to add my self hosted searxng to my firefox search engines I get an "Invalid Format" error with the following description:
`Firefox could not install the search engine from: https://searxng.mydomain.tld/opensearch.xml?method=POST&autocomplete=`
I accessed the URL in the error manually and checked the .xml file and it is valid and the entries seem correct. I have searxng installed from the official docker image but it's behind traefik instead of caddy, so maybe I need to do something different for it to work through traefik? Anyone have an idea?

**Comment by silmarine** [comment]
I figured out what was causing the issue. I have Authelia setup as a middleware for traefik. Even though I am logged into authelia on my firefox I guess when firefox requests the opensearch.xml it does it wtihout my authelia session cookie. When I set my searxng url to bypass in authelia everything works. I'd rather not do that though. If I bypass it just long enough to add the search it does still work but my autocomplete doesn't. Does anyone know the exact path that is being used to make autocomplete requests? I could possibly bypass just that resource in authelia.

**Reply by return42** [reply, marked as answer]
> Does anyone know the exact path that is being used to make autocomplete requests? I could possibly bypass just that resource in authelia.

    /autocompleter?q={searchTerms}

From the routing ..

https://github.com/searxng/searxng/blob/8762863ebfe7b00d4d07dde24e69112e9a605079/searx/webapp.py#L822

And here in the opensearch.xml (template) ..

https://github.com/searxng/searxng/blob/8762863ebfe7b00d4d07dde24e69112e9a605079/searx/templates/simple/opensearch.xml#L15-L17
